File: frontend/src/types/meter.ts
export interface TimeSeriesDataPoint {
  timestamp: string;
  actual: number;
  predicted: number | null;
  residual: number | null;
}

export interface TimeSeriesResponse {
  buildingNumber: string;
  utility: string;
  units: string;
  resolution: string;
  data: TimeSeriesDataPoint[];
}

File: backend/app/schemas/meter.py
from pydantic import BaseModel


class TimeSeriesDataPoint(BaseModel):
    timestamp: str
    actual: float
    predicted: float | None = None
    residual: float | None = None


class TimeSeriesResponse(BaseModel):
    buildingNumber: int
    utility: str
    units: str
    resolution: str
    data: list[TimeSeriesDataPoint]
